book.rb

```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental) unless @rentals.include?(rental)
+    rental.book = self
+  end
```

Comment:
- Great job done so far, but this does not really establish the `many-to-many` relationship between the `Rental` class and `Book` class. Please refer to this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for a better understanding. 👍 👍

Reply:
@mckent05 What are you talking about? your comment doesn't explain anything, I'm definitely achieving `many-to-many` relationship since I'm using an array of rentals in `book` and `person`, Rental is a mid class and it doesn't have many to many relationship with `person` or `book`. Please watch this diagram below for more information...

![image](https://user-images.githubusercontent.com/50754458/216135385-4856bfbd-4a1c-4b47-ac24-fe407f6e0245.png)


Rental has a relationship of one to many with both `person` and `book`....

Reply:
- Hello @wassimchakib, a `Rental` belongs to both the `Book` class and a `Person` class, therefore we need to ensure that when creating a new rental, we create the `belong-to` relationship between the above-mentioned classes. With what you have done, when adding a rental to a book, the person renting the book is missing.

But not to worry, if you still do not agree with my review, kindly use this [form](https://airtable.com/shrQAqnBwek5a0O0s) to request a second opinion, and a Senior code review will check out your work.